### DISCUSSION TITLE
XML Code Notes (data structure) : How to use ?

### DISCUSSION THREAD

**Discussion by PhilB-RLB** [TriliumNext/Trilium#2493] [Q&A] [answered]
Hello,
it's noticed here (https://github.com/zadam/trilium/wiki/Note) that "code note - some kind of formal code, typically programming language (e.g. JavaScript) or data structure (e.g. XML)" can be used in Trilium Notes.

I would like to use it to read some data for my music album collection. At this time, when I add a new album I select an artist from a list provided by a Promoted Attribute. This list show the artists last names in an alphabetical order :
...
Beatles (The)
Bowie, David
Clapton, Eric
...

Now, let says that I select the artist name "Bowie, David" from my dropdown list, how can I retrieve the following information from an XML Code note to obtain the following values in a script :
Artist First Name : David
Artist Last Name : Bowie

What would be the XML structure of note ?

The idea is to set a text for a new note with David Bowie (instead of the Promoted Attribute value "Bowie, David).

**Comment by zadam** [comment, marked as answer]
If you can choose then I recommend JSON since it's easier to work with in JavaScript.

Not sure how much you already know, but once you have a note object, then getting the underlying JSON data content is simple like this:

```javascript
const data = note.getJsonContent();

for (const artist of data) {
     console.log(artist.firstName, artist.lastName);
}
```

Structure can be whatever you want, e.g. array of objects as used in the code sample.

**Reply by PhilB-RLB** [reply]
Hello Zadam, once again thank you for your help.

Could you please provide the syntaxe to use in the JSON note in order to use you previous code example ?

**Reply by zadam** [reply]
```javascript
[
     {
            "firstName": "David",
            "lastName:" "Bowie"
     },
     {
            "firstName": "Serj",
            "lastName:" "Tankian"
     },
]
```

**Reply by PhilB-RLB** [reply]
Thank you very much Zadam !!!
